platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/IncrementalChangesContext.java

```diff
@@ -47,4 +59,9 @@ public ImmutableList<String> getRebuildReasons() {
     public Optional<ExecutionStateChanges> getChanges() {
         return Optional.ofNullable(executionStateChanges);
     }
+
+    @Nullable
+    public HashCode getCacheKey() {
+        return cacheKey;
+    }
```

Comment:
:x: This should return `Optional<HashCode>`.

Reply:
Done.